app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,85 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export_id)
+    ActiveRecord::Base.connection_pool.with_connection do
+      csv_export = CsvExport.find(csv_export_id)
+      create_export_folder
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # clean up downloaded files older than 12 hours or any jobs stuck for over 1 month
+    # TODO delete files with no record associations
+    end_date = Time.now - Rails.application.config.samson.export_job.downloaded_age
+    timeout_date = Time.now - Rails.application.config.samson.export_job.max_age
+    @csv_exports = CsvExport.where("(status = 'downloaded' AND updated_at <= :end_date) OR updated_at <= :timeout_date",
+      {end_date: end_date, timeout_date: timeout_date})
+    if @csv_exports
+      @csv_exports.each do |csv_downloaded|
+        filename = get_filename(csv_downloaded)
+        File.delete(filename) unless !File.exist?(filename)
+        csv_downloaded.delete
+        Rails.logger.info("Downloaded file #{csv_downloaded.filename} deleted")
+      end
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.filename
+    filename = get_filename(csv_export)
+
+    if csv_export.content == "deploys"
+      csv_export.started!
+      deploy_csv_export(filename)
+    else
+      csv_export.failed!
+    end
+
+    export_completed_notify(csv_export)
+
+    rescue
+
+    delete_file(get_filename(csv_export))
+    csv_export.failed!
+    Rails.logger.info("Export #{csv_export.filename} failed");
+  end
+  
+  def export_completed_notify(csv_export)
+    if csv_export.started?
+      CsvMailer.created_email(csv_export)
+      csv_export.finished!
+      Rails.logger.info("Export #{csv_export.filename} completed")
+    end
+  end
+  
+  def deploy_csv_export(filename)
+    @deploys = Deploy.joins(:stage).all
```

Comment:
In the original deploy export there was an issue that was causing the export to fail with deleted projects.  This was a quick fix to get it to work.
deploy 1, project a
deploy 2, project b (deleted) -> errors out

This was a simple way to make it skip over where projects were deleted and have a correct record count to see if the report failed midway in creation.

This issue shows itself in in the deploy views as well when it encounters a deploy record with a deleted project/stage.  For us deleted projects are so far back like page 5000+ that noone would run into it unless crawling.